src/useSkeleton.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { useMemo } from 'react'
+import type { SkeletonProps } from './Skeleton'
+import { Skeleton } from './Skeleton'
+
+const emptyProps: SkeletonProps = {}
+
+/**
+ * Returns `withSkeleton` HOC which conditionally returns a loading skeleton in place of contents
```

Comment:
Hmm, I don't think `withSkeleton` is really a HOC since it returns an **element** not a **component**. I'm trying to think of a better name for `withSkeleton` that doesn't make it sound like a HOC... maybe `skeletonOr`? As in `skeletonOr(<div />)`. Not sure if that is better.

Reply:
You're right: it is a monad.  Granted, that can sound quite intimidating to some.   Common terminoligy is to use `maybe()` in place of `withSkeleton()` - But I prefer the specificity of `withSkeleton`